docs/csi-controller-attach-detach.md

```diff
@@ -0,0 +1,128 @@
+  # Please refer to the top-level description of tolerations for an example
+  tolerations: []
+  # affinity provides more granular control of which kubernetes node will run the CSI controller server
+  # Please refer to the top-level description of affinity for an example
+  affinity: {}
+```
+The controller pod must be able to access the management endpoint of your Pure Storage backends. Other than that, there
+are no restrictions on where the controller server may run.
+
+### Database
+**WARNING:** The database values must be set *before* installing the plugin. We will not attempt to move database pods
+after startup except for in the case of failures.
+
+To restrict where the cockroach-operator, db-deployer, and pso-db pods may run, you may set these values:
+```yaml
+database:
+  # nodeSelector is the simplest way to limit which kubernetes nodes will run the database-related pods
+  # Please refer to the top-level description of nodeSelector for an example
+  nodeSelector: {}
+  # tolerations allows the database-related pods to run on tainted kubernetes nodes
+  # Please refer to the top-level description of tolerations for an example
+  tolerations: []
+  # affinity provides more granular control of which kubernetes nodes will run the database-related pods
+  # Please refer to the top-level description of affinity for an example
+  affinity: {}
+```
+The cockroach-operator, db-deployer, and pso-db pods must all be able to access the management endpoint of your Pure
+Storage backends. These nodes must also be time-synced to less than a 500 ms interval. Please also ensure that these
+nodes are able to sustain a small distributed database workload.
+
+## Examples
+### Restrict all REST-requiring pods to master nodes only
+```yaml
+# Allow node servers to run on all nodes (so that Pure PVCs can be mounted anywhere)
+nodeServer:
+  nodeSelector: {}
+  tolerations: []
+  affinity: {}
+
+controllerServer:
+  # Require that the controller server runs on control plane nodes
+  # WARNING: this label is an example as to how Rancher sets up master node labels.
+  # Your cluster may use different labels, so please update this example accordingly.
+  nodeSelector: {
+    "node-role.kubernetes.io/controlplane": "true"
+  }
+  # Allow the controller server to run on tainted nodes
+  tolerations:
+    - operator: Exists
+      effect: "NoSchedule"
+    - operator: Exists
+      effect: "NoExecute"
+  
+  # Affinity is not required since we already have a node selector
+  affinity: {}
+
+database:
+  # Require that the database pods run on control plane nodes
+  nodeSelector: {
+    "node-role.kubernetes.io/controlplane": "true"
+  }
+  # Allow the database pods to run on tainted nodes
+  tolerations:
+    - operator: Exists
+      effect: "NoSchedule"
+    - operator: Exists
+      effect: "NoExecute"
+  
+  # Affinity is not required since we already have a node selector
+  affinity: {}
+```
+
+### Restrict all PSO operations to one specific node
```

Comment:
Is this just one specific, or just specific nodes?

Is there not a label, like `worker` which is more generic to k8s we could use as the example?

Reply:
Surprisingly k8s has no predefined labels for master/worker/etcd that I could find. The only predefined labels are like... hostname, OS, and architecture (ARM/AMD64/etc.)

This would be restricting *all* PSO pods (database, controller, nodes) to specific labels, like if you had one subset of your cluster that had Pure connectivity but the rest had no storage connectivity/a different vendor or something. Kinda a strange use-case but I could see it at least.